distribution/lib/Standard/Base/0.0.0-dev/src/System/File.enso

```diff
@@ -51,11 +79,45 @@ new path = case path of
          import Standard.Base.System.File
          import Standard.Examples
 
-         example_read = File.read Examples.csv_path
-read : (Text | File) -> Text
-read path = .read <| case path of
-    Text -> (here.new path)
-    File _ -> path
+         example_read = File.read_text Examples.csv_path
+read_text : (Text | File) -> Encoding -> Problem_Behavior -> Text
+read_text path (encoding=Encoding.utf_8) (on_problems=Report_Warning) =
+    file = case path of
+        Text -> (here.new path)
+        File _ -> path
+        _ -> Error.throw (Illegal_Argument_Error "path should be either a File or a Text")
+    file.read_text encoding on_problems
+
+## ALIAS Write Text File
+
+   Open and write to the file at the provided `path`.
+
+   Arguments:
+   - path: The path of the file to open and read the contents of. It will
+     accept a textual path or a file.
+   - contents: The text to write to the file.
+   - encoding: The text encoding to decode the file with. Defaults to UTF-8.
+
+   ? Module or Instance?
+     If you have a variable `file` of type `File`, we recommend calling the
+     `.read_text` method on it directly, rather than using
+     `File.read_text file`. The later, however, will still work.
+
+   > Example
+     Read the `data.csv` file in the project directory's `data` directory. You
+     will need to create the file `data.csv` manually in that directory.
+
+         import Standard.Base.System.File
+         import Standard.Examples
+
+         example_read = File.read_text Examples.csv_path
```

Comment:
Given that we use the `Examples.csv_path` it will read `Enso_Project.data` from within the `Examples` project and IIRC `Enso_Project` resolves differently depending on the file in which it is called (always resolves to the project containing the executed file), so it will point to `data/...` inside of the Examples project and not in any user's projects. And so this `data.csv` file should probably already be there.